js/ffs.ts

```diff
@@ -27,15 +28,12 @@ function normalize(query) {
       const c1 = normalize_recursive(rem_query.queries[0]);
       const c2 = normalize_recursive(rem_query.queries[1]);
       // return cross product of c1 and c2
-      const c = [];
-      for (let i = 0; i < c1.length; i++)
-        for (let j = 0; j < c2.length; j++) {
-          c.push({
-            logical: "and",
-            queries: c1[i].queries.concat(c2[j].queries)
-          });
-        }
-      return c;
+      return c1.flatMap((c1i) =>
```

Comment:
This kills non chrome edge:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#browser_compatibility

Reply:
95% according to https://caniuse.com/mdn-javascript_builtins_array_flatmap

Reply:
I know. 
But this should be an explicit decision.